add $union support

maeldur opened the pull request:
Hello, this pull request adds $union support via people.union.  

Added test cases, and updated readme.md updated with examples.

tdumitrescu commented:
Many thanks for the patch; we'll take a look next week. Is this independent of the code referenced in https://github.com/mixpanel/mixpanel-node/issues/39?

maeldur commented:
yes, this is an independent patch

tdumitrescu commented on lib/mixpanel-node.js:
```diff
@@ -576,6 +576,60 @@ var create_client = function(token, config) {
+                var val = data[key];
+                if (util.isArray(val)) {
+                    $union[key] = val;
+                } else if (typeof(val) !== 'object') {
```
Everything looks great in this PR. The only issue I see is that this check is kind of overpermissive; it prevents sending objects as union vals but permits other types such as functions:
```js
mixpanel.people.union(distinct_id, {'woops': function() {}});
```
That's a contrived example, but some less obvious accidents will go through happily:
```js
mixpanel.people.union('bob', mixpanel);
// Sending the following data to Mixpanel (Engage):
// { '$union':
//    { token: [ 'mytoken' ],
//      send_request: [ [Function] ],
//      track: [ [Function] ],
//      import: [ [Function] ],
//      import_batch: [ [Function] ],
//      alias: [ [Function] ],
//      set_config: [ [Function] ] },
//   '$token': 'mytoken',
//   '$distinct_id': 'bob' }
```
Maybe just explicitly allowing only strings and numbers is the way forward?

maeldur commented:
Thanks for the feedback on the diff - you're right. I've updated the checks to allow only string/number values, or arrays of them (filtering others), and updated the test case. 

I wanted to ensure that users were getting what they expect if they tried  
```
mixpanel.people.union('13793', {'Items Purchased': ['socks', 'shoes']})
```
based on the official example for $union below (from https://mixpanel.com/help/reference/http) which uses an array value for merging
```
// This combines ["socks", "shirts"] with the existing values for the "Items purchased"
// list for user 13793, also ensuring that the list values will only appear once in
// the merged list.
{
    "$token": "36ada5b10da39a1347559321baf13063",
    "$distinct_id": "13793",
    "$union": { "Items purchased": ["socks", "shirts"] }
}
```

tdumitrescu commented:
Nice, looks good to go!
